packages/react-async/src/useAsync.js

```diff
@@ -20,6 +21,9 @@ const useAsync = (arg1, arg2) => {
   const lastOptions = useRef(undefined)
   const lastPromise = useRef(neverSettle)
   const abortController = useRef({ abort: noop })
+  const [preventLoop] = useState(loopPreventer)
```

Comment:
I'd use `useRef`instead of `useState` here and keep preventLoop in a ref, as it is not really application state. But that's more a thing of semantics and will not affect anything.

Maybe move it out in a custom hook like this:

```typescript
function useLoopPreventer() {
    const preventLoop = useRef();
    if (!preventLoop.current) { 
        preventLoop.current = loopPreventer(); 
    }
    return preventLoop.current;
}
```

Reply:
It can't be a hook, because `<Async>` is backwards compatible to React 16.3 which doesn't support hooks.

Reply:
The reason I used `useState` is for its lazy initial value. I give it a function that will only be invoked once, which is exactly what I want. It can be done with useRef but requires more code and doesn't take a resolver function for the initial value, so we'd be creating a new function each render, which we don't need.